### DISCUSSION TITLE
Discharge Missing for Some Years Despite Precipitation in wflow.jl Model

### DISCUSSION THREAD

**Discussion by AldenShadd1000** [Deltares/Wflow.jl#505] [Q&A]
I am encountering an issue in my wflow.jl model where discharge is not being generated for certain years, even though precipitation data is present during those periods. However, for other years with similar precipitation input, the model successfully calculates and outputs discharge.

Could anyone suggest potential reasons or solutions for this issue?

Any guidance or suggestions would be greatly appreciated.

Thanks.

**Comment by verseve** [comment]
Thanks for posting this question @AldenShadd1000 .

You are mentioning that discharge is not generated for certain years. Do you mean that discharge has zero values? Or at some point NaN values (I would also expect that the run at some point fails)? And do drying rivers occurs in the catchment? 

If the model runs fine and discharge is zero, and flow intermittence occurs, it is perhaps caused by higher potential evapotranspiration values?

**Reply by AldenShadd1000** [reply]
Thanks for your response @verseve .

Discharge doesn't have zero values nor any nan values, rather it has lower values from 1985 to 2005 when compared to discharge in 2006 to 2023, while precipitation pattern is similar in both periods. 

I have attached the following excel file of precipitation and discharge comparison and temperature for whole model time period (1981-2023) for better understanding.

Thanks and looking forward!

[Discharge_Precip_Temp_Comaparison.xlsx](https://github.com/user-attachments/files/17845148/Discharge_Precip_Temp_Comaparison.xlsx)

**Reply by verseve** [reply]
Indeed a strange pattern. What wflow version are you using? And is water demand included in your setup? What is the potential evapotranspiration (PET) pattern for the whole period? 

Difficult to say what might be the issue here. Besides checking the PET pattern, as a first step I would recommend to output gridded variables, for example variables precipitation, temperature, PET, river flow, overland flow and satwaterdepth for a part of both periods to check for possible causes.

**Reply by AldenShadd1000** [reply]
Wflow version that I am using is 1.10.4 (2024-06-04). Potential evapotranspiration for the whole period lies between 0.0 to 6.0 mm. Can you elaborate more on water demand?

I have thoroughly analyzed all the variables precipitation, temperature, PET for both periods and the their pattern is consistent and there are no anomalies in the data that could explain the issue. I have attached the forcing data file for your review to seek further clarification. Link for file: https://1drv.ms/u/s!AtArswnSDMnghHgLt_IWPJowgVrA?e=UahRHJ 

Subsequently, I ran the model for the extended period from 1963 to 2023 instead of the initial range of 1981 to 2023. Interestingly, the discharge calculations started only from 1988 to 2023, while the period from 1963 to 1987 showed previous phenomenon of flat discharge values. Can you please explain why this happened? Is there a more efficient approach to run the model without requiring a 25-year warm-up period, as it involves significant labor costs.

**Reply by verseve** [reply]
The version you are mentioning is the Julia version. Do you also know the wflow version? 

Did you setup the model with hydromt_wflow?

In the docs the water demand option is explained from [here](https://deltares.github.io/Wflow.jl/stable/model_docs/vertical/sbm/#Non-irrigation). Yes, as a first step checking the external forcing data is a good idea. However, I would also recommend to check the forcing variables and the other variables internally, as mentioned previously, to check if forcing variables are picked up correctly by wflow and if states/output variables can explain the pattern. Below an example to get gridded model output for some variables (TOML file):

```toml
[output]
path = "output.nc"

[output.vertical]
satwaterdepth = "satwaterdepth"
zi= "zi"
ustorelayerdepth = "ustorelayerdepth"
precipitation = "precipitation"
temperature = "temperature"
potential_evaporation = "potential_evaporation"

[output.lateral.river]
q_av = "q_av_river"

[output.lateral.land]
q_av = "q_av_land"
```

A 25 year warm-up period is not required, normally a 1-5 year period is sufficient.

And are reservoirs or lakes included in the catchment you're modelling? These can have a large impact on discharge patterns (e.g. these locations could have zero outflow when volume is small).

**Reply by AldenShadd1000** [reply]
Sorry for late response!

My Wflow version installed in julia is 0.8.1.

Yes, I did setup my model thorough Hydromt Wflow.

I have checked the forcing and other variables, files are correct. I"ll also print those above gridded variables to double verify it.

No, there are no lakes or reservoirs in our model.

**Reply by AldenShadd1000** [reply]
Hi @verseve, I have checked all the forcing parameters especially precipitation and PET, they are showing normal behaviour, but the discharge takes about 25-30 years to normalize, after those odd 25-30 years it gives normal results of q_river (discharge).

**Reply by vers-w** [reply]
Hi @AldenShadd1000 , did you also check (sensitive) model parameter values? For example, for another wflow model, that was built with ISRIC SoilGrids 2020 (and not 2017), `theta_s` and `theta_r` showed very low values, effectively resulting in very low soil storage values. Other soil related parameters to check: `kv_0`, `f`, `c` and `soilthickness`. And what kind of pattern is `satwaterdepth` showing over time? Based on the discharge timeseries you did share, I would expect `satwaterdepth` is declining at least for the first 25-30 years. 

You could also try to run with a warm state (`reinit = false`), for example with the state at the end of your run to check the effect on discharge.

**Reply by AldenShadd1000** [reply]
I am using ISRIC Soilgrids 2017 and soil related parameters are ok. satwaterdepth is continuously decreasing for first 6-7 years, then it is continuously increasing for almost next 15-20 years, after this period of odd 25 years, satwaterdepth doesn't continuously decline or rise, rather it varies in a range.

**Reply by vers-w** [reply]
And what about running with a warm state? If the discharge pattern for the first 25 years is state related, using a warm state should show a normal pattern.

**Reply by vers-w** [reply]
And could you please share the TOML file and log file of your simulation?

**Reply by AldenShadd1000** [reply]
When I try running with a warm state (reinit = false), it gives error on instates.nc, as I don't have the file, how can I produce it.

Please find attached toml and log file.
[log.txt](https://github.com/user-attachments/files/22133738/log.txt)
[wflow_sbm.txt](https://github.com/user-attachments/files/22133754/wflow_sbm.txt)

**Reply by vers-w** [reply]
For the `instates.nc` file you can just use the ouput state file `run_default/outstate/outstates.nc`, by copying this file to `instate/instates.nc`.

From the log.txt a couple of sensitive soil parameters are set to default values:

```
[ Info: Set `vertical.theta_s` using default value `0.6`.
[ Info: Set `vertical.theta_r` using default value `0.01`.
[ Info: Set `vertical.kv_0` using default value `3000.0`.
```
We did remove non-ASCII characters in v0.8.0 (`kv₀`, `θᵣ`, `θₛ`). In order to use the netCDF input variables, you need to rename these keys in the TOML file to `kv_0`, `theta_r` and `theta_s`. I also noticed that LAI data is not used (`leaf_area_index`), while this is part of the default setup of hydromt_wflow and recommended to include (e.g. `canopygapfraction` is now set to the default value of 0.1). When using LAI data `kext`, `specific_leaf` and `storage_wood` are also required (to compute `canopygapfraction` and `cmax`).

**Reply by AldenShadd1000** [reply]
I"ll try simulating model with with warm states. 

Thanks for updating on updated variables.

For LAI, I went on to this website :"https://lpdaac.usgs.gov/products/mcd15a3hv006/" according to the data catalog file but on website it says the data has been deprecated and no longer available.
<img width="1332" height="867" alt="image" src="https://github.com/user-attachments/assets/e48db8fb-ba54-4253-bd65-7a5327fc70be" />
Can you please help me out with lai dataset.
